cmake/ConfigureCompiler.cmake

```diff
@@ -228,7 +234,20 @@ else()
   endif()
   if (GCC)
     add_compile_options(-Wno-pragmas)
-
+    set(USE_AVX ON CACHE BOOL "Enable AVX instructions")
+    if (USE_AVX)
+      add_compile_options(-mavx)
+    else()
+      add_compile_options(-msse4)
+    endif()
+    # Intentionally using builtin memcpy.  G++ does a good job on small memcpy's when the size is known at runtime.
```

Comment:
Should this be "known at compile time" ?

Reply:
From observing it's behavior, it looks like G++ assumes it can always beat the library version of memcpy if it can deduce the value of size (like `memcpy(dest, src, sizeof(int) * 3)`).

Ideally, we could tell it not to bother if size is greater than some constant, but I don't see an option for that (in the long term, maybe we should just upgrade to a newer compiler).  

Anyway, we have to enable / disable it at compile time, since g++ will emit different code depending on what we want to do.

It might make sense to have the changes I made to flow.cpp conditionally link in depending on what sort of CPU we're running on, but the hope is that AVX2 is old enough for nobody to notice (it was introduced in 2013), though I added a cmake flag to limit to SSE (which is even older).